tests/rptest/tests/controller_snapshot_test.py

```diff
@@ -131,10 +132,21 @@ def __init__(self, redpanda, node):
             'internal_rpc_status', 'internal_rpc_port'
         ]
         self.brokers = dict()
-        for broker in admin.get_brokers(node=node):
+        brokers_resp = admin.get_brokers(node=node)
+        for broker in brokers_resp:
             self.brokers[broker['node_id']] = dict(
                 (k, broker.get(k)) for k in broker_fields)
 
+        # New brokers may not report the version yet, so gather the cluster version
+        # based on multiple brokers' reported version
+        versions = list(
+            set(broker['version'] for broker in brokers_resp
+                if 'version' in broker))
+        assert len(versions) == 1, f"Unexpected versions: {versions}"
+        version_line = versions[0]
+
+        self.version = ver_triple(version_line)
+
```

Comment:
Force-pushed to make the version uniformity check here more lenient by allowing missing versions from some nodes. This is to make the `test_join_restart_catch_up` test less flaky.